deployment/local/test_api.py

```
@@ -124,171 +140,231 @@ def test_batch_predictions():
+        print(f"❌ Batch prediction error: {e!s}")
         return False
 
+
 def test_rate_limiting():
     """Test rate limiting functionality."""
     print("\n5. Testing rate limiting...")
-    
+
     def make_request():
         try:
             response = requests.post(
```

Ruff (0.13.1)

Probable use of `requests` call without timeout

(S113)